app_dart/lib/src/model/commit_ref.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:github/github.dart';
+import 'package:meta/meta.dart';
+
+/// Represents components of a GitHub commit without specifying the backend.
+@immutable
+final class CommitRef {
```

Comment:
Nit / curious: instead of importing a upper level path and adding the "toRef()" methods on the other objects; why not have a named constructor (CommitRef.fromCommit()) or an extension?  The extension would keep the other changes in the PR the same.

Reply:
This PR gets rid of `.fromCommit` so you don't need an import something in order to convert:
![image](https://github.com/user-attachments/assets/a8db970b-29d0-42ff-a4ec-8f3dd4740b59)

I don't see the value in the extension, we own the type.